server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -1892,29 +1890,30 @@ public MNode getSeriesSchemasAndReadLockDevice(InsertPlan plan) throws MetadataE
   }
 
   public MeasurementMNode getOrCreateMeasurementMNode(
```

Comment:
IMO, This design needs to be reconsidered.
`public MeasurementMNode getOrCreateMeasurementMNode()` is a public methd and create a measurement node actually does not depend on an  `InsertPlan`. Instead, it only depends on `DataType`. Therefore, the method's parameter should be `TSDataType dataType`, rather than InsertPlan. (If you want to avoid code duplication, a private method may be needed).

About method `protected void autoCreateSchemaOrNot`: this method's logic is simply enough, why not just create a method `protected void autoCreateSchema` and move `autoCreateSchemaOrNot ` outside?

Reply:
Hi, @jixuan1989 . I've improved the codes again.